pkg/api/pluginproxy/access_token_provider.go

```diff
@@ -97,7 +99,34 @@ func (provider *accessTokenProvider) getAccessToken(data templateData) (string,
 		return "", err
 	}
 
-	expiresOnEpoch, _ := strconv.ParseInt(token.ExpiresOnString, 10, 64)
+	var expiresOnEpoch int64
+
+	// If expire-on field exists, use that
+	switch token.ExpiresOnValue.(type) {
+	case float64:
+		expiresOnEpoch = int64(token.ExpiresOnValue.(float64))
+	case string:
+		expiresOnEpoch, err = strconv.ParseInt(token.ExpiresOnValue.(string), 10, 64)
+		if err != nil {
+			logger.Warn("Could not parse ExpiresOn in access token response, %w", err)
+		}
+	default:
+		// otherwise use expire-in
+		var expiresIn int64
+		switch t := token.ExpiresIn.(type) {
+		case float64:
+			expiresIn = int64(token.ExpiresIn.(float64))
+		case string:
+			expiresIn, err = strconv.ParseInt(token.ExpiresIn.(string), 10, 64)
+			if err != nil {
+				return "", err
+			}
+		default:
+			logger.Warn("expires_in with type: %v not supported", reflect.TypeOf(t))
+		}
+		expiresOnEpoch = time.Now().Unix() + (expiresIn)
+	}
```

Comment:
Would like to try to simplify this code, not sure how adding suport for another field added 28 lines of new code.  

Maybe try to break it out into a separate function, and explore ways to simplify the logic